firefox-ios/Client/Application/AppDelegate.swift

```diff
@@ -268,6 +269,46 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // Since we only need the topSites data in the archiver, let's write it
         widgetManager?.writeWidgetKitTopSites()
     }
+
+    // MARK: Telemetry
+    private func sendStartupTelemetry() {
+        _ = profile.autofill.reopenIfClosed()
+        profile.autofill.listCreditCards(completion: { cards, error in
+            guard let cards = cards, error == nil else { return }
+            self.sendCreditCardsSavedAllTelemetry(numberOfSavedCreditCards: cards.count)
+        })
+
+        let searchController = UISearchController()
+        let loginsViewModel = PasswordManagerViewModel(
+            profile: profile,
+            searchController: searchController,
+            theme: LightTheme()
+        )
+        let dataSource = LoginDataSource(viewModel: loginsViewModel)
+        loginsViewModel.loadLogins(loginDataSource: dataSource)
+
+        loginsViewModel.queryLogins("") { logins in
+            self.sendLoginsSavedAllTelemetry(numberOfSavedLogins: logins.count)
+        }
+    }
+
+    private func sendLoginsSavedAllTelemetry(numberOfSavedLogins: Int) {
+        let savedLoginsExtra = [TelemetryWrapper.EventExtraKey.loginsQuantity.rawValue: Int64(numberOfSavedLogins)]
+        TelemetryWrapper.recordEvent(category: .information,
+                                     method: .foreground,
+                                     object: .loginsSavedAll,
+                                     extras: savedLoginsExtra)
+    }
+
+    private func sendCreditCardsSavedAllTelemetry(numberOfSavedCreditCards: Int) {
+        let savedCardsExtra = [
+            TelemetryWrapper.EventExtraKey.creditCardsQuantity.rawValue: Int64(numberOfSavedCreditCards)
+        ]
+        TelemetryWrapper.recordEvent(category: .information,
+                                     method: .foreground,
+                                     object: .creditCardSavedAll,
+                                     extras: savedCardsExtra)
+    }
```

Comment:
thanks for looking into this! what do you think about adding this in its own class / struct to decouple from the AppDelegate? Similar to how we have `WebviewTelemetry`, `SponsoredTileTelemetry`, `PrivateBrowsingTelemetry`

Reply:
can do, initially I thought that only 2 measly events won't clutter the class too much but I see how this can grow in time so I'll make it a separate class